### DISCUSSION TITLE
[Unstable] Switching to another game mode crashes the server, can't join or make a new server afterwards

### DISCUSSION THREAD

**Discussion by Efflorescized** [FakeFishGames/Barotrauma#14776] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

After trying to switch to a new game mode, the server crashed and put everyone back to the server list. After this happened, I couldn't make a server of my own, producing the same error message.

This video is the first encounter of that error, the host tried to switch to the missions game mode.
https://github.com/user-attachments/assets/c0f65067-ce29-44b2-8091-b5eaba348590

This video shows the same error message when trying to make a new server after the first video has already taken place, 
https://github.com/user-attachments/assets/8f6ae46d-58fb-4d31-b039-e50e0ed70692

### Reproduction steps

1. Start a multiplayer server
2. Play some PvP games
3. Switch to missions or possibly any other game mode

### Bug prevalence

Just once

### Single player or multiplayer?

Multiplayer hosted from the in-game menu (= using a listen server)

### Version

Other

### -

1.6.12.0

### Which operating system did you encounter this bug on?

Windows

### Relevant error messages and crash reports

```shell
Error while reading a message from server. 
Exception thrown while reading segment ClientList at position 2688. Previous segments: Segment { Identifier = SyncIds, Pointer = 32 }.
Inner exception: Object reference not set to an instance of an object.
   at Barotrauma.Networking.GameClient.ReadClientList(IReadMessage inc) in <DEV>\Barotrauma\BarotraumaClient\ClientSource\Networking\GameClient.cs:line 1981
   at Barotrauma.Networking.GameClient.<ReadLobbyUpdate>b__124_0(ServerNetSegment segment, IReadMessage inc) in <DEV>\Barotrauma\BarotraumaClient\ClientSource\Networking\GameClient.cs:line 2219
   at Barotrauma.Networking.SegmentTableReader`1.Read(IReadMessage msg, SegmentDataReader segmentDataReader, ExceptionHandler exceptionHandler) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Utils\SegmentTable.cs:line 296
```

**Comment by Jasontti** [comment]
Tested a fix for this, no issues found.
Closing discussion as resolved.
